file-attribute-caching/build.gradle.kts

```diff
@@ -0,0 +1,131 @@
+plugins {
+    kotlin("jvm")
+    `maven-publish`
+    signing
+    alias(libs.plugins.ksp)
+}
+
+val attributeCachingFilesystemVersion: String by project
+version = attributeCachingFilesystemVersion
+
+ksp {
+    arg("autoserviceKsp.verify", "true")
+}
+
+dependencies {
+    api(project(":forwarding"))
+    implementation(libs.auto.service.annotations)
+    ksp(libs.auto.service.ksp)
+
+    testImplementation(libs.commons.io)
+    testImplementation(libs.truth)
+    testImplementation(libs.junit.jupiter.params)
+    testImplementation(libs.jimfs)
+
+    testRuntimeOnly(libs.junit.jupiter.engine)
+}
+
+kotlin {
+    explicitApi()
+    jvmToolchain { languageVersion.set(JavaLanguageVersion.of(8)) }
+}
+
+// <editor-fold desc="Publishing and Signing">
+
+java {
+    withJavadocJar()
+    withSourcesJar()
+}
+
+publishing {
```

Comment:
I think my last bit of feedback might be consolidating some of the repeated boilerplate. 

There's little difference between the files.

```
josh_b@mkembp-jbqa1 attributeCachingFileSystem % diff file-attribute-caching/build.gradle.kts forwarding/build.gradle.kts 
5d4
<     alias(libs.plugins.ksp)
11,14d9
< ksp {
<     arg("autoserviceKsp.verify", "true")
< }
< 
16,20d10
<     api(project(":forwarding"))
<     implementation(libs.auto.service.annotations)
<     ksp(libs.auto.service.ksp)
< 
<     testImplementation(libs.commons.io)
23c13
<     testImplementation(libs.jimfs)
---
>     testImplementation(libs.mockito.kotlin)
33,34d22
< // <editor-fold desc="Publishing and Signing">
< 
131d118
< // </editor-fold>
```

I'm somewhat partial to the idea of combining the files using https://docs.gradle.org/current/userguide/plugins.html#sec:script_plugins. Something like `apply(from = "publishingConfig.gradle.kts")`, where the identical publishing sections use a shared script.

Just a thought. Open to ideas / feedback on my idea.

Reply:
this is resolved when i moved forwardingFileSystem back into its own package in the file-attribute-caching module